dev/com.ibm.ws.grpc_fat/fat/src/com/ibm/ws/fat/grpc/ServiceInterceptorTests.java

```diff
@@ -0,0 +1,230 @@
+
+        worldChannel = ManagedChannelBuilder.forAddress(grpcServer.getHostname(), grpcServer.getHttpDefaultPort()).usePlaintext().build();
+        worldServiceBlockingStub = GreeterGrpc.newBlockingStub(worldChannel);
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        worldChannel.shutdownNow();
+        grpcServer.stopServer("CWWKT0202W");
+    }
+
+    /**
+     * This method is used to set the server.xml
+     */
+    private static void setServerConfiguration(LibertyServer server,
+                                               String serverXML) throws Exception {
+        System.out.println("Entered set server config with xml " + serverXML);
+        if (!serverConfigurationFile.equals(serverXML)) {
+            // Update server.xml
+            LOG.info("ServiceInterceptorTests : setServerConfiguration setServerConfigurationFile to : " + serverXML);
+            server.setMarkToEndOfLog();
+            server.setServerConfigurationFile(serverXML);
+            server.waitForStringInLog("CWWKG0017I");
+            serverConfigurationFile = serverXML;
+        }
+    }
+
+    /**
+     * Test a single grpc service interceptor
+     *
+     * This test adds a grpc element without a grpc interceptor and then updates
+     * to a grpc element with a grpc interceptor. Make sure the interceptor does not run
+     * when not configured, and make sure it runs when configured.
+     *
+     * <grpc target="helloworld.Greeter" serverInterceptors="com.ibm.ws.grpc.fat.helloworld.service.HelloWorldServerInterceptor"/>
+     *
+     * The simple interceptor prints a message to the log.
+     *
+     * @throws Exception
+     *
+     **/
+    @Test
+    public void testSingleServerInterceptor() throws Exception {
+
+        // Update to a config file with a <grpc> element with no interceptor
+        LOG.info("ServiceInterceptorTests : testSingleServerInterceptor() : update the server.xml file to one with a </grpc> element with no interceptor");
+        setServerConfiguration(grpcServer, GRPC_ELEMENT);
+        grpcServer.waitForConfigUpdateInLogUsingMark(appName);
+
+        // Send a request to the HelloWorld service and check for a response
+        HelloRequest person = HelloRequest.newBuilder().setName("Leigh").build();
+        HelloReply greeting = worldServiceBlockingStub.sayHello(person);
+
+        //Make sure the reply has Leigh in it
+        assertTrue(greeting.getMessage().contains("Leigh"));
+        //Make sure the Interceptor was not called and did not log a message
+        String interceptorHasRun = grpcServer.verifyStringNotInLogUsingMark("com.ibm.ws.grpc.fat.helloworld.service.HelloWorldServerInterceptor has been invoked!",
+                                                                            SHORT_TIMEOUT);
+        if (interceptorHasRun != null) {
+            Assert.fail(c + ": server.xml with <grpc> element no interceptor ran when it should not have in " + STARTUP_TIMEOUT + "ms");
```

Comment:
`STARTUP_TIMEOUT` should be corrected to `SHORT_TIMEOUT`

Reply:
Done